### ISSUE TITLE
Intead of using x and y variables, sf::Vector2 should be used

### ISSUE THREAD

**Issue by TheClonerx** [thisisausernamerightplsacceptthis/fairanintsquare#11]
this leads to more readability and consistency

**Comment by TheOnlyWalrus** [comment]
Makes more sense, in some cases

**Comment by TheClonerx** [comment]
For example [here](https://github.com/thisisausernamerightplsacceptthis/fairanintsquare/blob/0b8386ec997df35cbb9459597159b9cf30d9502b/tiles.hpp#L39) and [here](https://github.com/thisisausernamerightplsacceptthis/fairanintsquare/blob/master/npc.hpp#L76)

**Comment by TheClonerx** [comment]
Also, adding the two variants it's redundant and makes unnecessary duplicated code

**Comment by TheOnlyWalrus** [comment]
So basically change it to (Vector2f vec) and then just change to (vec.x, vec.y) to reference the x and y

**Comment by TheClonerx** [comment]
Yes, also notice that `sf::Vector2f` it's just an alias for `sf::Vector2<float>`, and that there's also `sf::Vector2i` and `sf::Vector2u` for `int` and `unsigned int` respectively

**Comment by TheOnlyWalrus** [comment]
I will keep that in mind

**Comment by pixelsharmana** [comment]
If you are talking about physics.hpp, this issue was already resolved on the newer commits

**Comment by TheOnlyWalrus** [comment]
Tiles and npcs is what they tagged

**Comment by TheClonerx** [comment]
That's good, tho I was talking in general

**Comment by TheOnlyWalrus** [comment]
It would actually be easier to use

**Comment by pixelsharmana** [comment]
Right, I see some overloads missing indeed
